### ISSUE TITLE
Detect only one object on image

### ISSUE THREAD

**Issue by Aleksey1555** [AlexeyAB/darknet#6655]
Hello!
I trained yolov3 to detect some objects, but it detects only one object on image.
maybe i missing something?

**Comment by stephanecharette** [comment]
No, you're missing something.  If the network is well-trained, it is definitely possible to find hundreds of objects within an image.  We'd need to know more about what you did to comment further.  Also consider joining the discord, you might get some further help there:  https://discord.gg/zSq8rtW

Here is an example from a YOLOv3-tiny network I trained last year.  In this image it has found 82 objects:

![image](https://user-images.githubusercontent.com/5061352/93005042-fdcc0480-f501-11ea-8816-e947aeae9fc0.png)

**Comment by stephanecharette** [comment]
> In training process i used images with only one object. And txt file with only one line.
> Can it be the reason?

You edited your question while I was replying!  Yes, this is one of the problems.  The network can only find things similar to what it was trained to find.  If you want it to find lots of object XYZ in an image, then train with images that have many XYZ, and make sure *all* of them are marked.  See here:  https://www.ccoderun.ca/darkmark/ImageMarkup.html

**Comment by Aleksey1555** [comment]
thanks for the answer!
I will try

**Comment by Aleksey1555** [comment]
Hello stephanecharette.
How you achive this results?
What about background? It always gray color or not?

**Comment by Aleksey1555** [comment]
Where you get images?
And how many images need to learn yolov3 to recognize digits?

**Comment by stephanecharette** [comment]
Background doesn't matter.  If you have enough marked up images, the network learns what is important, and ignores the rest.  See here for a few more examples:  https://www.ccoderun.ca/programming/ml/mailbox.html

I forget how many images and markup was needed to get the project started.  Because of the repetitiveness and similarities between images, it wasn't much.  Looking at the project now, it contains 374 images and has 14774 labels.  My first dataset in that project contains 70 images, and I think I only had half of it marked up the first time I trained.

I suggest you join the Darknet/YOLO discord if you want more help:  https://discord.gg/zSq8rtW

**Comment by stephanecharette** [comment]
As for "where you get images" -- I walked around the city and took pictures of all the mailboxes I could find.

**Comment by Aleksey1555** [comment]
Cool)
but what about color of the digits? is it matter? 
I train digits which handwritten by white color and black background. Is this means that my network will see only white digits?
Or color doesn't matter too?

**Comment by stephanecharette** [comment]
No, like I wrote, see the link above.  You can tweak the colour settings in your .cfg file if you want more colour changes in your data augmentation.  Though if you have the images and can do stuff like negative (inverse colour) it probably would help.
